content/blog/blog-2020-06-29-kuttl-v0.5.0.md

```diff
@@ -0,0 +1,139 @@
+---
+date: 2020-06-29
+---
+
+# KUTTL v0.5.0 Released
+
+Today we release KUTTL v0.5.0.  Most releases are simply announced, however this v0.5.0 release is so jam packed that it warrents a blog post to help those using kuttl to come up to speed on the latest features.  This post will outline the **highlights** of the v0.5.0 release which includes:
+
+* Assert and Errors Command
+* Namespace Control
+* Command Expansion with commands, and TestStep annotations
+* Pod Log Collector 
+* Test Reports
+
+<!-- more -->
+
+## Assert and Errors Commands
+
+Added to the KUTTL set of commands is `assert` and `errors`.  The premise behind these commands is while you are writing a new kuttl test, it is common to manually adjust or query the cluster for information as you form the assert or errors files.  It is super helpful to "check" an assert file against the current state of the cluster without the need to walk through the entire test steps for that test.  It could also be useful to assert a cluster to be ready for testing a set of tests. 
+
+As an example, perhaps you have a pod running and you want to confirm it is running.  A common kuttl parital yaml file might looking like (01-assert.yaml):
+
+```yaml
+apiVersion: v1
+kind: Pod
+metadata:
+  name: static-web
+status:
+  phase: Running
+```
+
+Running a `kubectl kuttl assert --timeout 5 01-assert.yaml` might provide an successful response such as:
+`"01-assert.yaml" in "default" namespace is valid`
+or it may provide a verbose output which ends with the details such as:
+
+`resource Pod:default/static-web: .status.phase: value mismatch, expected: Running != actual: Pending`
+
+::: warning
+These examples expect that you have a running kubernetes cluster and that kuttl can connect to it.
+:::
+
+The `errors` command provides the absolute affect, it asserts that the value doe NOT exist in the cluster. Example:
+```
+kubectl kuttl errors --timeout 5 01-assert.yaml
+error assert is valid
+```
+
+::: tip
+These commands work with URLs as well.  Example: `kubectl kuttl errors https://gist.githubusercontent.com/kensipe/ccf11977a1f9816865beae1a6e4e7883/raw/df0f0432387486410b5a3da20b5d45fead9e91b0/assert.yaml --timeout 1
+error assert is valid`
+:::
+
+
+## Namespace Control
+
+Previous to v0.5.0 KUTTL would always create a test namespace.  This works great in environments where you have full control of your cluster, however many orgs were unable to use kuttl for this reason allow.  It is now possible to pass a namespace to kuttl to use.  When a namespace is provided, kuttl will use that namespace if it exists and attempts to create if it doesn't.  The justification for creation is the test suite would have failed if the namespace didn't exist, if the kuttl user isn't authorized to create a namespace it will also fail. 
```

Comment:
```suggestion
Previous to v0.5.0 KUTTL would always create a test namespace.  This works great in environments where you have full control of your cluster, however many orgs were unable to use kuttl for this reason.  It is now possible to pass a namespace for kuttl to use.  When a namespace is provided, kuttl will use that namespace if it exists and attempts to create if it doesn't. The justification for creation is the test suite would have failed if the namespace didn't exist, if the kuttl user isn't authorized to create a namespace it will also fail. 
```